opensquirrel/ir.py

```diff
@@ -214,8 +214,29 @@ def __init__(self, *axis: AxisLike) -> None:
         axis: An ``AxisLike`` to create the axis from.
         """
         axis_to_parse = axis[0] if len(axis) == 1 else cast(AxisLike, axis)
+        if not self.has_valid_value(axis_to_parse):
+            raise ValueError
         self._value = self._parse_and_validate_axislike(axis_to_parse)
 
+    @staticmethod
+    def has_valid_value(*axis: AxisLike) -> bool:
+        """Check if the axis has a valid value.
+
+        Args:
+            axis: An ``AxisLike`` to validate.
+
+        Returns:
+            True if the axis is valid, False otherwise.
+        """
+        try:
+            axis_to_check = axis[0] if len(axis) == 1 else cast(AxisLike, axis)
+            axis_array = np.asarray(axis_to_check, dtype=float).flatten()
+            if len(axis_array) != 3:
+                return False
+            return not np.all(axis_array == 0)
```

Comment:
It's usually preferable to use the positive form, e.g., here:

```suggestion
            return np.any(axis_array != 0)
```

Reply:
Wouldn't any(axis != 0) mean that none of the values on the Axis can be zero? For example I believe that (1, 0, 1) would be a valid Axis, but it would not pass np.any(axis_array != 0)

Reply:
No, no, `any(x != 0)` means that "any of the values in Axis should be zero".

For "none of the values in Axis can be zero", i.e., "all of the values in Axis should be non-zero", you would have to write `all(x != 0)`.